src/buttons/button.ts

```diff
@@ -1,18 +1,26 @@
-import {Directive, Input, ElementRef, Renderer2} from '@angular/core';
+import {Component, ElementRef, Input, Renderer2} from '@angular/core';
 import {replaceClass} from '../util/util';
+import {ButtonVariantOptions} from '../util/types';
 
-@Directive({
-  selector: '[nglButton]',
+@Component({
+  selector: '[ngl-button]',
+  templateUrl: './button.pug',
 })
 export class NglButton {
 
-  private _type: string;
-  @Input() set nglButton(type: 'success' | 'neutral' | 'brand' | 'destructive' | 'inverse') {
-    replaceClass(this, `${this.prefix}${this._type}`, type ? `${this.prefix}${type}` : '');
-    this._type = type;
+  @Input() set variant(variant: ButtonVariantOptions) {
+    replaceClass(this, `${this.prefix}${this._variant}`, variant ? `${this.prefix}${variant}` : '');
+    this._variant = variant;
   }
+  @Input() iconName: string;
+  @Input() iconPosition: 'left' | 'right';
 
-  private prefix = `slds-button--`;
+  get svgClasses() {
+    return `slds-button__icon slds-button__icon_${this.iconPosition}`;
+  }
+
+  private _variant: ButtonVariantOptions;
```

Comment:
I prefer not to use external types, but inline them, as this will help documentation export later.